File: app/src/main/java/hu/pe/biko/biko/RouteActivity.java
package hu.pe.biko.biko;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

public class RouteActivity extends AppCompatActivity {
    Route route;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_route);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        route = getIntent().getParcelableExtra("route");
        TextView textView = (TextView) findViewById(R.id.text_route)
        textView.setText(route.getDescription());
        getSupportActionBar().setTitle(route.getName());

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(view -> {
//                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
//                        .setAction("Action", null).show();
            Intent intent = new Intent(RouteActivity.this, MapsActivity.class);
            intent.putExtra("route", route);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/hu/pe/biko/biko/RouteActivity.kt
package hu.pe.biko.biko

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide

class RouteActivity : AppCompatActivity() {
    internal var route : Route? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_route)
        val toolbar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)

        route = intent.getParcelableExtra<Route>("route")
        val textView = findViewById(R.id.text_route) as TextView
        textView.text = route!!.description
        supportActionBar!!.title = route!!.name
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        val fab = findViewById(R.id.fab) as FloatingActionButton
        fab.setOnClickListener { view ->
            //                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
            //                        .setAction("Action", null).show();
            val intent = Intent(this@RouteActivity, MapsActivity::class.java)
            intent.putExtra("route", route)
            startActivity(intent)
        }
        Glide.with(this).load(route!!.image)
                .error(R.drawable.caption)
                .placeholder(R.drawable.caption)
                .into(findViewById(R.id.image_route) as ImageView)
    }
}